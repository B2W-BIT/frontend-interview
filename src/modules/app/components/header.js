import React, {Component, PropTypes} from 'react'
import '../styles/header.scss'


export const Header = ({
    profile = {},
    fetching = false,
    fetched = false,
    error = null,
    children,
}) => {
    return (
        <header className="hidden-sm hidden-xs">
            <div className="custom-nav">
                <div className="custom-nav__inner">
                    <div className="container">
                        <div className='pull-left'>
                            <a href='#'> <span className='Icon -bird'></span> Home</a>
                            <a href='#'> <span className='Icon -raio'> </span>Moments </a>
                        </div>
                        <div className='pull-right'>
                            <input className='search' type='text' name='search' placeholder='Buscar' />
                            <span className="login caret">VocÃª possui uma conta? Entrar.</span>
                        </div>
                    </div>
                </div>
            </div>

        </header>
    )
}
