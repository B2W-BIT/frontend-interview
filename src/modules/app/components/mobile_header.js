import React, {Component, PropTypes} from 'react'
import '../styles/mobile_header.scss'


export const MobileHeader = ({}) => {
    return (
        <header className="mobile-header hidden-lg hidden-md flex" role="banner">
            <div className="flex">
                <div className="mobile-header__icons-row mobile-header__row flex">
                    <div className="mobile-header__row-content flex">
                        <div className="mobile-header__row-align flex">
                            <div className="mobile-header__icon -icon-left flex">
                            <i className="Icon -bird flex"></i>
                            </div>
                            <div className="mobile-header__spacing"></div>
                            <div className="mobile-header__icon -icon-right flex">
                            <i className="Icon -search flex"></i>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="mobile-header__row flex">
                    <div className="mobile-header__row-content flex -bottom">
                        <div className="mobile-header__row-align flex">
                            <div className="mobile-header__btn-wrapper flex">
                                <div className="flex-row flex">
                                    <div className="mobile-header__btn flex">
                                        <button className="btn btn-default -primary">Inscrever-se</button>
                                    </div>
                                    <div className="mobile-header__btn flex">
                                        <button className="btn btn-default">Entrar</button>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </header>
    )
}
