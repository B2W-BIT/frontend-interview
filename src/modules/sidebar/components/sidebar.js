import React, {Component, PropTypes} from 'react'
import '../styles/sidebar.scss'

import { timeUtils, twitterHelpers } from '../../../helpers'
import verifiedIcon from '!svg-inline!../svg/verified.svg'

const parseProfilePicUrl = url => {
    return url.replace('normal', 'reasonably_small')
}

export const Sidebar = ({
    name,
    verified,
    screen_name,
    description,
    location,
    url,
    friends_count,
    followers_count,
    created_at,
    profile_image_url,
    followersYouKnow,
    mediaPosts
}) => {
    created_at = new Date(created_at)
    return (
        <div className="sidebar">
            <div className="sidebar__profile-pic">
                <img src={ parseProfilePicUrl(profile_image_url) } />
            </div>
            <div className="sidebar__name-container">
                <h1>
                    <div>
                        <span className="sidebar__name">{ name }</span>
                        <span className="sidebar__verified">
                        {
                            verified && (<span dangerouslySetInnerHTML={{ __html: verifiedIcon }} />)
                        }
                        </span>
                    </div>
                    <small className="sidebar__screen_name">
                        @{ screen_name }
                    </small>
                </h1>

            </div>
            <div className="sidebar__description">
                { description }
            </div>
            <div className="sidebar__row">
                {
                    !!location && (<div className="sidebar__location">
                        { location }
                    </div>)
                }
                 <div>-</div>
                {
                    !!url && (<div className="sidebar__site">
                        <a href={ url }>{ url } </a>
                    </div>)
                }
            </div>

            <div className="sidebar__row hidden-lg hidden-md">
                {
                    !!friends_count && (
                        <div className="sidebar__following">
                            <strong>{ twitterHelpers.parseStats(friends_count) }</strong> SEGUINDO
                        </div>
                    )
                }
                {
                    !!followers_count && (
                        <div className="sidebar__followers-count">
                            <strong>{ twitterHelpers.parseStats(followers_count) }</strong> SEGUIDORES
                        </div>
                    )
                }

            </div>
            <div className="hidden-sm hidden-xs">
                {
                    !!created_at && (
                        <div className="sidebar__created_at">
                            Participa desde { timeUtils.monthName(created_at.getMonth()) } de { created_at.getFullYear() }
                        </div>
                    )
                }
                <div className="sidebar__followersYouKnow">
                    {
                        followersYouKnow(screen_name)
                    }
                </div>
                <div className="sidebar__media">
                    {
                        mediaPosts(screen_name)
                    }
                </div>
            </div>

        </div>
    )
}
