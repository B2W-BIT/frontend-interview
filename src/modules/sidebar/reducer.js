// import { } from './constants'

const initialState = {
}

export function sidebar(state = initialState, action) {

    switch (action.type) {

        default:
            return state
    }
}
