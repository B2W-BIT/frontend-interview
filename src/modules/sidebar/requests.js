import { Config } from '../../config'
import Axios from 'axios'

const { apiUrl } = Config.url
