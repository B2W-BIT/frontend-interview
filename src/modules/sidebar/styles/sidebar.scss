.sidebar {
    position: relative;
    margin-top: 23px;
    padding: 20px;
    background-color: #fff;
    &__profile-pic {
        width: 86px;
        height: 86px;
        position: absolute;
        left: 3%;
        top: -25%;
        img {
            width: 100%;
        }
    }
    &__name-container {
        margin-top: 5%;
        h1 {
            line-height: 0;
        }
    }
    &__screen_name {
        font-size: 0.99999rem;
    }
    &__name {
        font-size: 1.285714286rem;
        font-weight: bold;
        line-height: 0;

    }
    &__verified {
        font-size: 1.285714286rem;
    }
    &__row {
        div {
            display: inline-block;
            width: auto;
            margin: 1%;
        }
        div:first-child {
            margin-left: 0;
        }
    }
}
