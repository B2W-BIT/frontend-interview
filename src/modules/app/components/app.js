import React, {Component, PropTypes} from 'react'
import '../styles/app.scss'

import { Header } from './header.js'
import { MobileHeader } from './mobile_header.js'

//
export const App = ({
    children
}) => {
    return (
        <div className="flex">
            <Header />
            <MobileHeader />
            <main className="flex" role="main">
                {  children }
            </main>
            <div className="footer"></div>
        </div>
    )
}
