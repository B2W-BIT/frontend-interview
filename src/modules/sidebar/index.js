/*Vendor imports*/
import React, {Component, PropTypes} from 'react'
import { connect } from 'react-redux'
/*Custom imports*/

import { Sidebar } from './components/sidebar'

class SidebarContainer extends Component {

    constructor(props) {
        super(props)
    }

    followersYouKnow(screen_name) {
        return ""
    }

    mediaPosts(screen_name) {
        return ""
    }

    render() {
        return (
            <Sidebar {...this.props} followersYouKnow={this.followersYouKnow} mediaPosts={this.mediaPosts} />
        )
    }
}

const mapStateToProps = state => ({ ...state.users })

const mapDispatchToProps = dispatch => {
    return {
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(SidebarContainer)
