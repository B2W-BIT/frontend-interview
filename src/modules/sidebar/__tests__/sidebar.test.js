import React from 'react';
import { Sidebar } from '../components/sidebar'
import renderer from 'react-test-renderer';

describe("Will test the sidebar component", () => {

    it('renders de cover photo nicely', () => {
      const props = {
          name: "",
          verified: "",
          screen_name: "",
          description: "",
          location: "",
          profile_image_url: "",
          url: "",
          friends_count: "",
          followers_count: "",
          created_at: "",
          followersYouKnow: () => "",
          mediaPosts: () => ""
      }
      const tree = renderer.create(
        <Sidebar { ...props } />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

})
