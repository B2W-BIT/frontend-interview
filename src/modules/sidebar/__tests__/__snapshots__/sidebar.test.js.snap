exports[`Will test the sidebar component renders de cover photo nicely 1`] = `
<div
  className="sidebar">
  <div
    className="sidebar__profile-pic">
    <img
      src="" />
  </div>
  <div
    className="sidebar__name-container">
    <h1>
      <div>
        <span
          className="sidebar__name">
          
        </span>
        <span
          className="sidebar__verified">
          
        </span>
      </div>
      <small
        className="sidebar__screen_name">
        @
        
      </small>
    </h1>
  </div>
  <div
    className="sidebar__description">
    
  </div>
  <div
    className="sidebar__row">
    <div>
      -
    </div>
  </div>
  <div
    className="sidebar__row hidden-lg hidden-md" />
  <div
    className="hidden-sm hidden-xs">
    <div
      className="sidebar__created_at">
      Participa desde 
       de 
      NaN
    </div>
    <div
      className="sidebar__followersYouKnow">
      
    </div>
    <div
      className="sidebar__media">
      
    </div>
  </div>
</div>
`;
